flax/experimental/nnx/nnx/graph.py

```diff
@@ -1072,11 +1072,11 @@ def _iter_nodes(
   if id(node) in visited:
```

Comment:
Please add a docstring to `iter_nodes` explaining the iteration order, since that function is now public.